### PR TITLE
NXBT-1915: remove old Git repositories parent folders

### PR DESCRIPTION
**Pull request by jcarsique**
Two new features:
- remove Git repositories parent folders older than 5 days
- remove Git repositories parent folders older than 2 days and bigger than 100M

Update header and document each line

### CODE REVIEW

**Comment by alexistimic** [comment]
:+1:

File: jenkins_workspace_cleanup.sh [hunk 1]
```diff
@@ -1,30 +1,38 @@
...
 docker volume prune -f > /dev/null
+# Delete "exited" containers
 docker ps -a | awk '/Exited.*(hours|days|weeks) ago/ {print $1}' | xargs --no-run-if-empty docker rm -v > /dev/null 2>&1
+
 find /opt/jenkins/workspace/TestAndPush -maxdepth 1 -mindepth 1 -type d -mtime +3 -exec rm -rf {} +
 find '/opt/jenkins/workspace*/' -name 'nuxeo-server-tomcat-*.zip' -o -name 'nuxeo-cap-*.zip' -mtime +3 -exec rm {} +
 find '/opt/jenkins/workspace*/' -path '*/target/tomcat' -type d -prune -mtime +3 -exec rm -r {} +
+
+# Remove Git repositories parent folders older than 5 days
+find /opt/jenkins/workspace* -maxdepth 2 -type d -execdir test -d {}/.git \; -mtime +5 -prune -print -exec rm -r -- {} +
```
Comment (efge) [inline jenkins_workspace_cleanup.sh]: Please use quotes around arguments that contain wildcards that must not be interpreted by the shell: `'/opt/jenkins/workspace*'`

Comment (jcarsique) [reply jenkins_workspace_cleanup.sh]: we want them being interpreted:

    # find '/opt/jenkins/workspace*'
    find: '/opt/jenkins/workspace*': No such file or directory

Comment (efge) [reply jenkins_workspace_cleanup.sh]: Ok then I misunderstood, and the ones above must be changed back too (and use a common syntax for trailing slashes please)

File: jenkins_workspace_cleanup.sh [hunk 2]
```diff
@@ -1,30 +1,41 @@
...
 docker ps -a | awk '/Exited.*(hours|days|weeks) ago/ {print $1}' | xargs --no-run-if-empty docker rm -v > /dev/null 2>&1
+
+# Delete T&P jobs older than 3 days
 find /opt/jenkins/workspace/TestAndPush -maxdepth 1 -mindepth 1 -type d -mtime +3 -exec rm -rf {} +
-find '/opt/jenkins/workspace*/' -name 'nuxeo-server-tomcat-*.zip' -o -name 'nuxeo-cap-*.zip' -mtime +3 -exec rm {} +
-find '/opt/jenkins/workspace*/' -path '*/target/tomcat' -type d -prune -mtime +3 -exec rm -r {} +
+
+# Delete Nuxeo server ZIP files and unzipped folders older than 3 days
+find /opt/jenkins/workspace*/ -name 'nuxeo-server-tomcat-*.zip' -o -name 'nuxeo-cap-*.zip' -mtime +3 -exec rm {} +
+find /opt/jenkins/workspace*/ -path '*/target/tomcat' -type d -prune -mtime +3 -exec rm -r {} +
```
Comment (efge) [inline jenkins_workspace_cleanup.sh]: I also note mixed usage, `rm -r -- {}` in some cases and `rm -r {}` in others. `--` should probably always be present.

**Comment by efge** [comment]
👍

**Pull request merged**